libraries/rush-sdk/src/helpers.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, FileSystem } from '@rushstack/node-core-library';
+import type { EnvironmentVariableNames } from '@microsoft/rush-lib';
+
+export const RUSH_LIB_NAME: '@microsoft/rush-lib' = '@microsoft/rush-lib';
+export const RUSH_LIB_PATH_ENV_VAR_NAME: typeof EnvironmentVariableNames.RUSH_LIB_PATH = '_RUSH_LIB_PATH';
+
+export type RushLibModuleType = Record<string, unknown>;
+
+export class SdkContext {
+  public rushLibModule: RushLibModuleType | undefined = undefined;
+}
+
+export const sdkContext: SdkContext = new SdkContext();
+
+/**
+ * Find the rush.json location and return the path, or undefined if a rush.json can't be found.
+ *
+ * @privateRemarks
+ * Keep this in sync with `RushConfiguration.tryFindRushJsonLocation`.
+ */
+export function tryFindRushJsonLocation(startingFolder: string): string | undefined {
+  let currentFolder: string = startingFolder;
+
+  // Look upwards at parent folders until we find a folder containing rush.json
+  for (let i: number = 0; i < 10; ++i) {
+    const rushJsonFilename: string = path.join(currentFolder, 'rush.json');
+
+    if (FileSystem.exists(rushJsonFilename)) {
```

Comment:
Ideally use `fs.existsSync` to avoid taking a dependency on the (very expensive) node-core-library.

Reply:
I agree that eliminating `node-core-library` makes sense for `rush-sdk`, since we want the proxy to be as slim as possible.  But that's a more involved change and I think out of scope for this PR.